src/pages/dataElementGroups/fields/ShortNameField.tsx

```diff
@@ -0,0 +1,62 @@
+import i18n from '@dhis2/d2-i18n'
+import { InputFieldFF } from '@dhis2/ui'
+import React, { useMemo } from 'react'
+import { Field as FieldRFF, useField } from 'react-final-form'
+import { useParams } from 'react-router-dom'
+import {
+    composeAsyncValidators,
+    required,
+    useCheckMaxLengthFromSchema,
+    useIsFieldValueUnique,
+} from '../../../lib'
+import type { SchemaName } from '../../../types'
+import type { FormValues } from '../form'
+
+function useValidator() {
+    const params = useParams()
+    const dataElementId = params.id as string
+    const checkIsValueTaken = useIsFieldValueUnique({
+        model: 'dataElements',
```

Comment:
`dataElementGroups`. Please consider extracting this.

Reply:
Here's a quick example of how I think this could look like when generalized:

```ts


function useValidator() {
    const params = useParams()
    const schemaSection = useSchemaSectionHandleOrThrow()

    const modelId = params.id as string
    const checkIsValueTaken = useIsFieldValueUnique({
        model: schemaSection.namePlural,
        field: 'name',
        id: modelId,
    })

    const checkMaxLength = useCheckMaxLengthFromSchema(
        schemaSection.name,
        'formName'
    )

    return useMemo(
        () =>
            composeAsyncValidators<string, FormValues>([
                checkIsValueTaken,
                checkMaxLength,
                required,
            ]),
        [checkIsValueTaken, checkMaxLength]
    )
}

type ShortNameFieldProps = {
    helpText?: string
    required?: boolean
}
export function ShortNameField({ helpText, required }: ShortNameFieldProps) {
    const validator = useValidator()
    const { meta } = useField('shortName', {
        subscription: { validating: true },
    })

    const requiredLabel = i18n.t('{{fieldLabel}} (required)', {
        fieldLabel: i18n.t('Short name'),
    })

    const label = required ? requiredLabel : i18n.t('Short name')

    return (
        <FieldRFF<string | undefined>
            loading={meta.validating}
            component={InputFieldFF}
            dataTest="formfields-shortname"
            required
            inputWidth="400px"
            label={label}
            name="shortName"
            helpText={helpText}
            validate={(name?: string) => validator(name)}
            validateFields={[]}
        />
    )
}


```